ganttproject/src/biz/ganttproject/impex/csv/GanttCSVExport.java

```diff
@@ -102,25 +97,84 @@ public void save(OutputStream stream) throws IOException {
       format = format.withQuote(csvOptions.sSeparatedTextChar.charAt(0));
     }
 
-    CSVPrinter csvPrinter = new CSVPrinter(writer, format);
+    return format;
+  }
+
+  private SpreadsheetWriter getWriter(OutputStream stream, String formatName) throws IOException {
+    if (formatName != null && formatName.length() > 0) {
```

Comment:
Let me remind that `formatName` is not format name now, it is file extension. Why don't you replace it with enum value? This will eliminate checks and `IllegalArgumentException`s -- your compiler will check that you don't pass anything wrong here.